File: data_import.m
kill
wd = '/home/kln/projects/geertz/mystic_anthro/code';
%% vector space
% ## data
dd = '/home/kln/projects/geertz/mystic_anthro/data';
cd(dd)
% import data
data = load('av_gz_data.mat');
dtm_gz = data.dtm_gz;
% geertz
file_gz = csv2cell('geertz_file.txt','fromfile');
    file_gz = strrep(file_gz(2:end,:),' ',''); % remove row and strip whitespace
voc_gz = csv2cell('geertz_voc.txt','fromfile');
    voc_gz = strrep(voc_gz(2:end,:),' ','');
% avila
dtm_av = data.dtm_av;
file_av = csv2cell('avila_file.txt','fromfile');
    file_av = strrep(file_av(2:end,:),' ','');
voc_av = csv2cell('avila_voc.txt','fromfile');
    voc_av = strrep(voc_av(2:end,:),' ','');
clear data
%save('vspc_data.mat')
%%
kill
% metadata for avila and geertz
load('/home/kln/projects/geertz/mystic_anthro/data/vspc_data.mat')
load('/home/kln/projects/geertz/mystic_anthro/data/avila_yr.mat')
tmp = file_av(:,2);
meta_av = cell(length(tmp),2);
for i = 1:length(tmp)
    meta_av(i,:) = strsplit(tmp{i},'_');
    meta_av{i,1} = str2num(meta_av{i,1}); %#ok<ST2NM>
end
tmp = file_gz(:,2);
meta_gz = cell(length(tmp),2);
meta_gz(:,1) = num2cell(1:length(tmp))';
for i = 1:length(tmp)
    %meta_gz(i,:) = strsplit(tmp{i},'_');
    l = regexp(tmp{i},'\d+', 'match'); 
    meta_gz{i,2} = str2num(cell2mat(l(end))); %#ok<ST2NM>
end
% concatenate and sort on year for meta and meta_yr for av
meta_mat_col = {'id' 'sort_id' 'sort_yr'};% column names
[y,i] = sort(meta_av_yr(:,2));
class_av = meta_av(i,2); % class information sorted on year
meta_av_mat = [cell2mat(meta_av(:,1)) cell2mat(meta_av(i,1)) y];
tmp = cell2mat(meta_gz);
[y, i ] = sort(tmp(:,2));
meta_gz_mat = [cell2mat(meta_gz(:,1)) i y];
% check sorting result
subplot(2,1,1), plot(meta_av_mat(:,3)); subplot(2,1,2), plot(meta_gz_mat(:,3));
%save('metadata.mat','meta_gz_mat','meta_av_mat','meta_mat_col')

File: data_import.R
rm(list = ls())
wd <- '/home/kln/projects/geertz/mystic_anthro/code'
dd <- '/home/kln/projects/geertz/mystic_anthro/data'
setwd(wd)
# packages
library(tm)
# functions
preprocess <- function(v.cor){
  tmp <- tm_map(v.cor, PlainTextDocument)
  tmp <- tm_map(tmp, removePunctuation)
  tmp <- tm_map(tmp, removeNumbers)
  tmp <- tm_map(tmp, content_transformer(tolower))
  tmp <- tm_map(tmp, removeWords, stopwords("english"))
  tmp <- tm_map(tmp, stemDocument)
  tmp <- tm_map(tmp, stripWhitespace)
  cl.cor <- tm_map(tmp, PlainTextDocument)
  return(cl.cor)
}
docsparse <- function(mindocs,dtm){
  n = length(row.names(dtm))
  sparse <- 1 - mindocs/n;
  dtmreduce <- tm::removeSparseTerms(dtm, sparse)
  return(dtmreduce)
}
# data directory (vanilla utf-8)
dpath_1 <- "/home/kln/projects/geertz/mystic_anthro/data/geertz_data/geertz_txt"
dpath_2 <- "/home/kln/projects/geertz/mystic_anthro/data/avila_data/avila_txt"
#datadir <- "/home/kln/projects/human_futures/data/sandbox"
geertz.cor <- Corpus(DirSource(dpath_1, encoding  = "UTF-8"), readerControl = list(language="PlainTextDocument"))
avila.cor <-  Corpus(DirSource(dpath_2, encoding  = "UTF-8"), readerControl = list(language="PlainTextDocument"))
# import document names
file_1 <- gsub("\\..*","",names(geertz.cor))
file_2 <- gsub("\\..*","",names(avila.cor))
# preprocess quick and dirty
geertz.cl.cor <- preprocess(geertz.cor)
avila.cl.cor <- preprocess(avila.cor)
# vectspc
mindoc = floor(.05*length(file_1))# minimum rep based on smallest corpus
geertz.mat <- as.matrix(docsparse(mindoc,DocumentTermMatrix(geertz.cl.cor)))
rownames(geertz.mat) <- file_1
avila.mat <- as.matrix(docsparse(mindoc,DocumentTermMatrix(avila.cl.cor)))
rownames(avila.mat) <- file_2
# save work space
save.image(file = paste(dd,'/data_cor_mat.RData',sep = ""))

### export
#write.table(geertz.mat, file = paste(dd,'/geertz_mat.csv',sep = ""), row.names = FALSE, col.names = FALSE)
#write.table(avila.mat, file = paste(dd,'/avila_mat.csv',sep = ""), row.names = FALSE, col.names = FALSE)
write.table(rownames(geertz.mat), file = , paste(dd,'/geertz_file.txt',sep = ""))
write.table(colnames(geertz.mat), file = , paste(dd,'/geertz_voc.txt',sep = ""))
write.table(rownames(avila.mat), file = , paste(dd,'/avila_file.txt',sep = ""))
write.table(colnames(avila.mat), file = , paste(dd,'/avila_voc.txt',sep = ""))

library('R.matlab')
filename <- paste(dd,'/av_gz_data.mat', sep = "")
writeMat(filename, dtm_gz = geertz.mat, dtm_av = avila.mat)






